app/src/main/java/com/example/animationstestapp/ui/notifications/NotificationsViewModel.kt

```diff
@@ -1,13 +1,54 @@
 package com.example.animationstestapp.ui.notifications
 
+import android.app.Application
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.animationstestapp.data.model.Person
+import kotlinx.coroutines.launch
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.io.File
 
-class NotificationsViewModel : ViewModel() {
+class NotificationsViewModel(application: Application) : AndroidViewModel(application) {
 
-    private val _text = MutableLiveData<String>().apply {
-        value = "This is notifications Fragment"
+    private val _person: MutableLiveData<Person> = MutableLiveData<Person>(Person())
+
+    val person: LiveData<Person> get() = _person
+
+    private val resDir
+        get() = getApplication<Application>().applicationContext
+            .getDir(RES_DIR_FILE_NAME, Context.MODE_PRIVATE)
+
+    private val file: File
+        get() = File(resDir, PERSON_DATA_FILE_NAME)
+
+    fun saveChangesInFile(person: Person) {
+        if (file.exists())
+            viewModelScope.launch {
+                val json = Json.encodeToString(person)
+                file.writeText(json)
+                Log.d(TAG, "saveChangesInFile: File:$file Data:$json")
+            } else
+            Log.e(TAG, "saveChangesInFile: $file", NullPointerException("Data file is not set"))
+    }
+
+    fun loadData() {
+        if (file.exists()) {
```

Comment:
this logic should be incapsulated in you repository. ViewModel just takes the repository as a blackBox and everything is going there. Repository should return something like here https://betterprogramming.pub/how-to-use-kotlin-sealed-classes-for-state-management-c1cfb81abc6a